src/pal/src/misc/miscpalapi.cpp

```diff
@@ -235,92 +228,73 @@ PAL_GetPALDirectoryA(
     return bRet;
 }
 
-BOOL
+VOID
 PALAPI
 PAL_Random(
-        IN BOOL bStrong,
         IN OUT LPVOID lpBuffer,
         IN DWORD dwLength)
 {
-    int rand_des = -1;
-    BOOL bRet = FALSE;
+    static volatile int rand_des = -1;
     DWORD i;
-    char buf;
     long num = 0;
-    static BOOL sMissingDevRandom;
     static BOOL sMissingDevURandom;
     static BOOL sInitializedMRand;
 
     PERF_ENTRY(PAL_Random);
-    ENTRY("PAL_Random(bStrong=%d, lpBuffer=%p, dwLength=%d)\n", 
-          bStrong, lpBuffer, dwLength);
-
-    i = 0;
+    ENTRY("PAL_Random(lpBuffer=%p, dwLength=%d)\n", lpBuffer, dwLength);
 
-    if (bStrong == TRUE && i < dwLength && !sMissingDevRandom)
+    if (!sMissingDevURandom)
     {
-        // request non-blocking access to avoid hangs if the /dev/random is exhausted
-        // or just simply broken
-        if ((rand_des = PAL__open(RANDOM_DEVICE_NAME, O_RDONLY | O_NONBLOCK)) == -1)
+        if (rand_des == -1)
         {
-            if (errno == ENOENT)
-            {
-                sMissingDevRandom = TRUE;
-            }
-            else
+            int fd;
+
+            do
             {
-                ASSERT("PAL__open() failed, errno:%d (%s)\n", errno, strerror(errno));
+                fd = open("/dev/urandom", O_RDONLY, O_CLOEXEC);
```

Comment:
I do not think `PAL_Random` is used for anything perf critical in the runtime itself. I do not think that the caching is worth it here.

Reply:
Isn't any CoCreateGuid call in the runtime perf sensitive? I really cannot tell in some of the cases.
And is it worth having a slightly different implementation in corefx and coreclr just to save one open handle? I've answered your question on the count of open handles in simple hello world app - it is 21 without my change.

Reply:
If CoCreateGuid is used for anything perf sensitive, then the 8x perf difference between Windows and Unix is still quite a bit of a problem.

Reply:
The same actually true for the framework side of the implementation. 8x is still better than 16x compared to Windows, but you probably do not want to be calling it in too often in any case.

I guess if we wanted to make this better, we would need to cache (read say 256 bytes at a time) or have a local random number generator (with larger seed than the 32-bits of built-in one).